text/0000-delete-load.md

```diff
@@ -0,0 +1,48 @@
+# Replace frozen trajectory from file
+
+## Summary
+[summary]: #summary
+
+We propose to add `PoseGraphInterface::DeleteTrajectory`, `PoseGraphInterface::GetTrajectoryStates`, and `MapBuilderInterface::LoadStateFromFile` because these functions are missing to replace a frozen trajectory ("map") at run-time.
+
+## Motivation
+[motivation]: #motivation
+
+Currently, it is not possible to replace a finished (and frozen) trajectory at run-time.
+In fact, the trimmers are the only components that even delete parts of the pose graph.
+When we add these two functions to the public API, the use case of swapping a frozen map at run-time is possible.
+A caller can load a new frozen map, wait for global SLAM to process it (and re-localize in the new map), and delete the old frozen map.
+Another use case is to clean up old trajectories on a device that runs continuously and sometimes starts and finishes trajectories.
```

Comment:
I think the deletion can be even implicit for trajectories that have been finished and trimmed completely. Related to https://github.com/googlecartographer/cartographer/issues/1111

Reply:
I'm not sure if implicitly deleting is the right way to fix #1111.
Should we support `SerializeState` for a selection of trajectories in a future RFC?

Reply:
Yes improving the (selective) serialization would be something separate, this just came to my mind again when I read it.